homeassistant/components/sonos/media_browser.py

```diff
@@ -105,6 +106,22 @@ def media_source_filter(item: BrowseMedia) -> bool:
     return item.media_content_type.startswith("audio/")
 
 
+def _get_title(id_string: str) -> str:
+    """Extract a suitable title from the content id string."""
+    try:
+        if id_string.startswith("S:"):
+            # Format is S://server/share/folder
+            # If just S: this will be in the mappings; otherwise use the last folder in path.
+            title = LIBRARY_TITLES_MAPPING.get(
+                id_string, urllib.parse.unquote(id_string.split("/")[-1])
+            )
+        else:
+            title = urllib.parse.unquote(id_string.split("/")[1])
+    except IndexError:
```

Comment:
Do we have the IndexError because of the list lookups we do? Do we have better ways of detecting this?

Reply:
good suggestion; that was a remnant from the code snippet I factored into this function. I rewrote it do not rely on the indexerror.